pkg/storagepoolclaim/v1alpha1/storagepoolclaim.go

```diff
@@ -339,3 +341,26 @@ func (l *SPCList) GetPoolUIDs() []string {
 	}
 	return uids
 }
+
+// EstimateSPCVersion returns the csp version if any csp is present for the spc or
+// returns the maya version as the new csp created will be of maya version
+func (spc *SPC) EstimateSPCVersion() (string, error) {
+	var v string
+	cspList, err := csp.KubeClient().List(
+		metav1.ListOptions{
+			LabelSelector: string(apis.StoragePoolClaimCPK) + "=" + spc.Object.Name,
+		})
+	if err != nil {
+		return "", errors.Wrapf(
+			err,
+			"failed to get csplist for %s to add version details",
```

Comment:
nit: `"failed to get the csp list related to spc : %s",` ? 
As error should be why is errored.
In caller side we can return err adding  `" failed to add version details"`

Reply:
Done